lib/soap.js

```diff
@@ -526,10 +526,9 @@ class SOAPError extends Error {
         }
```

Comment:
i would suggest to add `this.name = this.constructor.name` into the constructor. according to https://futurestud.io/tutorials/node-js-create-your-custom-error this would prefix error output with - in this case - "SOAPError" instead of with "Error"

also, apparently it's good to include `Error.captureStackTrace(this, this.constructor);` to ensure the stack trace is proper.

Reply:
checked mozilla and they confirm the name & captureStackTrace idea: https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Error#custom_error_types